presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java

```diff
@@ -140,6 +144,38 @@ public void testCompact()
     }
 
     @Test
+    public void testCompactBlocks()
+            throws Exception
+    {
+        DictionaryId commonSourceId = randomDictionaryId();
+
+        Slice[] expectedValues = createExpectedValues(5);
+        // Both blocks do not reference value 2, only block2 references value 4
+        DictionaryBlock dictionaryBlock1 = createDictionaryBlockWithUnreferencedKeys(expectedValues, 10, commonSourceId);
```

Comment:
I think this test will be easier to read of you just list out the positions here
```java
DictionaryBlock dictionaryBlock1 = createDictionaryBlock(expectedValues, new int[] {0, 1, 1, 3, 3, 0, 1, 1, 3, 3}, commonSourceId);